### PR TITLE
TEZ-4282: Possible NPE in LocalClient after TEZ-4236

### PR DESCRIPTION
**Pull request by abstractdog**

### CODE REVIEW

File: tez-dag/src/main/java/org/apache/tez/client/LocalClient.java [hunk 1]
```diff
@@ -438,7 +447,9 @@ public DAGStatus apply(Set<StatusGetOpts> statusOpts, Long timeout) {
   public boolean shutdownSession(Configuration configuration, ApplicationId sessionAppId,
       UserGroupInformation ugi) throws TezException, IOException, ServiceException {
     if (isLocalWithoutNetwork) {
-      clientHandler.shutdownAM();
+      if (clientHandler != null){
+        clientHandler.shutdownAM();
+      }
       return true;
     } else {
```
Comment (pgaref) [inline tez-dag/src/main/java/org/apache/tez/client/LocalClient.java:454]: If - else can be simplified here (else not needed)

File: tez-dag/src/main/java/org/apache/tez/client/LocalClient.java [hunk 2]
```diff
@@ -403,7 +405,14 @@ protected DAGAppMaster createDAGAppMaster(ApplicationAttemptId applicationAttemp
   @Override
   public TezAppMasterStatus getAMStatus(Configuration configuration, ApplicationId appId,
       UserGroupInformation ugi) throws TezException, ServiceException, IOException {
-    return clientHandler.getTezAppMasterStatus();
+    if (isLocalWithoutNetwork) {
+      if (clientHandler == null) {
+        return TezAppMasterStatus.INITIALIZING;
+      }
+      return clientHandler.getTezAppMasterStatus();
+    } else {
```
Comment (pgaref) [inline tez-dag/src/main/java/org/apache/tez/client/LocalClient.java:413]: if - else can be simplified (no need for else.
Are we sure we need the super. getAMStatus call? I would assume that a FrameworkClient would call its own method right?

**Review by pgaref — COMMENTED** [review]
Hey @abstractdog  looks fine in general left some comments

File: tez-dag/src/main/java/org/apache/tez/client/LocalClient.java (hunk 1 above)
Comment (abstractdog) [reply tez-dag/src/main/java/org/apache/tez/client/LocalClient.java:454]: right, updating it

File: tez-dag/src/main/java/org/apache/tez/client/LocalClient.java (hunk 2 above)
Comment (abstractdog) [reply tez-dag/src/main/java/org/apache/tez/client/LocalClient.java:413]: else simplification - OK
yes we need it, this is called from:
https://github.com/apache/tez/blob/master/tez-api/src/main/java/org/apache/tez/client/TezClient.java#L834

if we don't want to use LocalClient's own logic for this codepath, we shall fallback to the parent's, which is FrameworkClient. getAMStatus (as LocalClient is a FrameworkClient)

Comment (pgaref) [reply tez-dag/src/main/java/org/apache/tez/client/LocalClient.java:413]: Oh I see, this part of the TEZ-4236 addition -- makes sense

Comment (abstractdog) [reply tez-dag/src/main/java/org/apache/tez/client/LocalClient.java:413]: thanks!

**Review by pgaref — APPROVED** [review]
LGTM otherwise +1 from my side

**Pull request merged**